tests/components/imap_email_content/test_repairs.py

```diff
@@ -0,0 +1,302 @@
+def mock_client() -> Generator[MagicMock, None, None]:
+    """Mock the imap client."""
+    with patch(
+        "homeassistant.components.imap_email_content.sensor.EmailReader.read_next",
+        return_value=None,
+    ), patch("imaplib.IMAP4_SSL") as mock_imap_client:
+        yield mock_imap_client
+
+
+CONFIG = {
+    "platform": "imap_email_content",
+    "name": "Notifications",
+    "server": "imap.example.com",
+    "port": 993,
+    "username": "john.doe@example.com",
+    "password": "**SECRET**",
+    "folder": "INBOX.Notifications",
+    "value_template": "{{ body }}",
+    "senders": ["company@example.com"],
+}
+DESCRIPTION_PLACEHOLDERS = {
+    "yaml_example": ""
+    "template:\n"
+    "- sensor:\n"
+    "  - name: Notifications\n"
+    "    state: '{{ trigger.event.data[\"text\"] }}'\n"
+    "  trigger:\n  - event_data:\n"
+    "      sender: company@example.com\n"
+    "    event_type: imap_content\n"
+    "    id: custom_event\n"
+    "    platform: event\n",
+    "server": "imap.example.com",
+    "port": 993,
+    "username": "john.doe@example.com",
+    "password": "**SECRET**",
+    "folder": "INBOX.Notifications",
+    "value_template": '{{ trigger.event.data["text"] }}',
+    "name": "Notifications",
+}
+
+CONFIG_DEFAULT = {
+    "platform": "imap_email_content",
+    "name": "Notifications",
+    "server": "imap.example.com",
+    "port": 993,
+    "username": "john.doe@example.com",
+    "password": "**SECRET**",
+    "folder": "INBOX.Notifications",
+    "senders": ["company@example.com"],
+}
+DESCRIPTION_PLACEHOLDERS_DEFAULT = {
+    "yaml_example": ""
+    "template:\n"
+    "- sensor:\n"
+    "  - name: Notifications\n"
+    "    state: '{{ trigger.event.data[\"subject\"] }}'\n"
+    "  trigger:\n  - event_data:\n"
+    "      sender: company@example.com\n"
+    "    event_type: imap_content\n"
+    "    id: custom_event\n"
+    "    platform: event\n",
+    "server": "imap.example.com",
+    "port": 993,
+    "username": "john.doe@example.com",
+    "password": "**SECRET**",
+    "folder": "INBOX.Notifications",
+    "value_template": '{{ trigger.event.data["subject"] }}',
+    "name": "Notifications",
+}
+
+
+@pytest.mark.parametrize(
+    ("config", "description_placeholders"),
+    [
+        (CONFIG, DESCRIPTION_PLACEHOLDERS),
+        (CONFIG_DEFAULT, DESCRIPTION_PLACEHOLDERS_DEFAULT),
+    ],
+    ids=["with_value_template", "default_subject"],
+)
+async def test_deprecation_repair_flow(
+    hass: HomeAssistant,
+    mock_client: MagicMock,
+    hass_client: ClientSessionGenerator,
+    hass_ws_client: WebSocketGenerator,
+    config: str | None,
+    description_placeholders: str,
+) -> None:
+    """Test the deprecation repair flow."""
+    # setup config
+    await async_setup_component(hass, "sensor", {"sensor": config})
+    await hass.async_block_till_done()
+
+    state = hass.states.get("sensor.notifications")
+    assert state is not None
+
+    await async_process_repairs_platforms(hass)
```

Comment:
Why do we need to do this explicitly?

Reply:
Copied this from test code for integration, but it can do without